velox/expression/tests/CastExprTest.cpp

```diff
@@ -171,6 +171,86 @@ class CastExprTest : public functions::test::CastBaseTest {
     assertEqualVectors(expected, result);
   }
 
+  template <typename T>
+  void testDecimalToIntCasts() {
+    auto shortFlat = makeNullableShortDecimalFlatVector(
+        {-300, -230, -200, -100, 0, 5'500, 5755, 6'900, 7'200, std::nullopt},
+        DECIMAL(6, 2));
+    testComplexCast(
+        "c0",
+        shortFlat,
+        makeNullableFlatVector<T>(
+            {-3,
+             -2 /*-2.3 rounds to -2*/,
+             -2,
+             -1,
+             0,
+             55,
+             57 /*57.55 rounds to 57*/,
+             69,
+             72,
+             std::nullopt}));
+    auto longFlat = makeNullableLongDecimalFlatVector(
+        {-30'000'000'000,
+         -25'500'000'000,
+         -20'000'000'000,
+         -10'000'000'000,
+         0,
+         550'000'000'000,
+         559'900'000'000,
+         690'000'000'000,
+         720'000'000'000,
+         std::nullopt},
+        DECIMAL(20, 10));
+    testComplexCast(
+        "c0",
+        longFlat,
+        makeNullableFlatVector<T>(
+            {-3,
+             -2 /*-2.55 rounds to -2*/,
+             -2,
+             -1,
+             0,
+             55,
+             55 /* 55.99 rounds to 55*/,
+             69,
+             72,
+             std::nullopt}));
+
+    if constexpr (!std::is_same<int64_t, T>::value) {
+      const auto tooSmall = (int64_t)std::numeric_limits<int32_t>::min() - 1;
+      const auto tooBig = (int64_t)std::numeric_limits<int32_t>::max() + 1;
+
+      VELOX_ASSERT_THROW(
+          testComplexCast(
+              "c0",
+              makeShortDecimalFlatVector({tooSmall}, DECIMAL(9, 0)),
+              makeFlatVector<T>(0)),
+          "Failed to cast from DECIMAL(9,0)");
```

Comment:
It would be nice to include the 'to' type in the error message and the reason for failure.

Reply:
Done.